XenModel/Actions/HealthCheck/XenServerHealthCheckUpload.cs

```diff
@@ -214,5 +229,12 @@ public string UploadZip(string fileName, string caseNumber, System.Threading.Can
             log.InfoFormat("Succeeded to upload bundle {0}", fileName);
             return uploadUuid;
         }
+
+        #region IDisposable Members
+        public void Dispose()
+        {
+            httpClient.Dispose();

```

Comment:
The HttpClientHandler is missing before httpClient (it will have to be promoted from local variable to class field).

Reply:
The HttpClientHandler is disposed automatically by HttpClient.Dispose(), because the HttpClient constructor used, sets the internal flag `disposeHandler` to true by default. See the constructor implementation in the HttpClient class:
``` 
    /// <summary>Initializes a new instance of the <see cref="T:System.Net.Http.HttpClient" /> class with a specific handler.</summary>
    /// <param name="handler">The HTTP handler stack to use for sending requests. </param>
    [__DynamicallyInvokable]
    public HttpClient(HttpMessageHandler handler)
      : this(handler, true)
    {
    }

    /// <summary>Initializes a new instance of the <see cref="T:System.Net.Http.HttpClient" /> class with a specific handler.</summary>
    /// <param name="handler">The <see cref="T:System.Net.Http.HttpMessageHandler" /> responsible for processing the HTTP response messages.</param>
    /// <param name="disposeHandler">
    /// <see langword="true" /> if the inner handler should be disposed of by Dispose(), <see langword="false" /> if you intend to reuse the inner handler.</param>
    [__DynamicallyInvokable]
    public HttpClient(HttpMessageHandler handler, bool disposeHandler)
      : base(handler, disposeHandler)
    {
    ....
```
However, as this is not clear in the code, I will change it to explicitily use the second constructor, `HttpClient(HttpMessageHandler handler, bool disposeHandler)`.

Reply:
Fair enough.